### PR TITLE
Fixes #16728: rl enchance variables handling 

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/16728

wip don't merg, waiting for last PR merging and still some fixes incoming

### CODE REVIEW

**Comment by gpoblon** [comment]
Commit modified
variable tests added + parameters translate output adjustment

**Comment by gpoblon** [comment]
In the end all changes in this PR are specifically related to parameters rather than variables
Can now be merged

File: rudder-lang/libs/generate-lib [hunk 1]
```diff
@@ -27,6 +27,7 @@ foreach my $file (glob $ncf_methods) {
   # only one paramater, TODO hack for translate
   if ($file =~ /\/variable_string_escaped/) { next; }
 
+variable = { parameter_count=2 }
```
Comment (peckpeck) [inline rudder-lang/libs/generate-lib]: Why is this here ?

**Comment by peckpeck** [comment]
you need to rebase

**Comment by gpoblon** [comment]
PR updated with a new commit
deleted line that had nothing to do here

File: rudder-lang/libs/generate-lib (hunk 1 above)
Comment (gpoblon) [reply rudder-lang/libs/generate-lib]: fixed

**Comment by peckpeck** [comment]
could you rebase this one ?

**Comment by gpoblon** [comment]
PR rebased

File: rudder-lang/src/parser.rs [hunk 2]
```diff
@@ -599,9 +599,24 @@ fn pstruct(i: PInput) -> PResult<HashMap<String, PValue>> {
     )(i)
 }
 
+/// Alternative version of pstruct based on "." rather than braces.
+/// Used for the agent
+/// A struct is stored in a HashMap
+// fn pstruct_agent(i: PInput) -> PResult<HashMap<String, PValue>> {
```
Comment (peckpeck) [inline rudder-lang/src/parser.rs]: What is this ?

**Comment by gpoblon** [comment]
Commit modified

File: rudder-lang/src/parser.rs (hunk 2 above)
Comment (peckpeck) [reply rudder-lang/src/parser.rs]: I think you should remove this

**Comment by gpoblon** [comment]
Commit modified
was in next PR but removed it anyway

File: rudder-lang/src/parser.rs (hunk 2 above)
Comment (gpoblon) [reply rudder-lang/src/parser.rs]: was removed from the next PR but I just removed it here too

File: rudder-lang/src/parser.rs [hunk 3]
```diff
@@ -766,6 +784,52 @@ fn pvariable_definition(i: PInput) -> PResult<(Token, PValue)> {
     )(i)
 }
 
+fn pvalue_agent(i: PInput) -> PResult<PValue> {
```
[the remark this answers is not in the document]
Comment (gpoblon) [reply rudder-lang/src/parser.rs]: This content has been removed and is now part of another PR

**Comment by peckpeck** [comment]
OK, merging this PR
